testutil/src/main/java/org/spine3/test/NullToleranceTest.java

```diff
@@ -0,0 +1,500 @@
+            }
+
+            final boolean primitive = allPrimitiveTypes().contains(key);
+            if (primitive) {
+                result = defaultValue(key);
+            }
+
+            final boolean wrapper = isWrapperType(key);
+            if (result == null && wrapper) {
+                final Class<?> unwrappedPrimitive = unwrap(key);
+                result = defaultValue(unwrappedPrimitive);
+            }
+
+            final Class<GeneratedMessageV3> messageClass = GeneratedMessageV3.class;
+            final boolean message = messageClass.isAssignableFrom(key);
+            if (result == null && message) {
+                result = getDefaultMessageInstance(key);
+            }
+
+            final Class<String> stringClass = String.class;
+            if (result == null && stringClass.isAssignableFrom(key)) {
+                result = STRING_DEFAULT_VALUE;
+            }
+
+            checkState(result != null);
+            return result;
+        }
+
+        private static Object getDefaultMessageInstance(Class<?> key) {
+            try {
+                final Method method = key.getMethod(METHOD_NAME, EMPTY_PARAMETER_TYPES);
+                final Object defaultInstance = method.invoke(null, EMPTY_ARGUMENTS);
+                return defaultInstance;
+            } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+                throw Exceptions.wrappedCause(e);
+            }
+        }
+    }
+
+    /**
+     * A builder for producing the {@link NullToleranceTest} instance.
+     */
+    public static class Builder {
+
+        private final Set<String> excludedMethods;
+        private final Map<? super Class, ? super Object> defaultValues;
+        private Class targetClass;
+
+        private Builder() {
+            defaultValues = newHashMap();
+            excludedMethods = newHashSet();
+        }
+
+        /**
+         * Sets the target class.
+         *
+         * @param utilClass the utility {@link Class}
+         * @return the {@code Builder}
+         */
+        public Builder setClass(Class utilClass) {
+            this.targetClass = checkNotNull(utilClass);
+            return this;
+        }
+
+        /**
+         * Adds the method name which will be excluded from the check.
+         *
+         * @param methodName the name of the method to exclude
+         * @return the {@code Builder}
+         */
+        @SuppressWarnings("WeakerAccess") // it is a public API.
+        public Builder excludeMethod(String methodName) {
+            checkNotNull(methodName);
+            final boolean validName = isName(methodName);
+            checkArgument(validName);
+
+            excludedMethods.add(methodName);
+            return this;
+        }
+
+        /**
+         * Adds the default value for the method argument.
+         *
+         * <p> Each static and non-private method in the {@code targetClass} executes during the check.
```

Comment:
```
<p> During the checks, each applicable method of the target class is executed with the {@code null} values passed as arguments. This is done in an iterative fashion, so that each invocation checks the {@code null} tolerance for the only one of the parameters. The rest of the method arguments are set with the default values.

<p>If the default value for the type is not customized, the predefined list of the default values per type will be used:

(here goes the list of the default values per type)

<p><b>Example.</b>

<p>The method declared as

<pre> {@code public static void doSomething(PersonName messageValue,
                                                                              Integer wrapperValue,
                                                                              CustomType objectValue)} </pre>

<p> will be invoked three times:

<ol> 
	<li> {@code doSomething(null, 0, <default value for CustomType>)},
	<li> {@code doSomething(PersonName.getDefaultInstance(), null , <default value for CustomType>)},
	<li> {@code doSomething(PersonName.getDefaultInstance(), 0 , null)}.
</ol> 

<p> If the default value for the {@code CustomType} is not provided, an {@code IllegalStateException} is thrown.
```